xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-preferences/xwiki-platform-notifications-preferences-api/src/main/java/org/xwiki/notifications/preferences/script/NotificationPreferenceScriptService.java

```diff
@@ -286,13 +287,35 @@ public NotificationEmailDiffType getDiffType()
     }
 
     /**
-     * @param userId id of a user
+     * @param user reference of a user
      * @return the diff type for emails configured for the given user
-     * @since 9.11RC1
+     * @since 14.10
+     */
+    @Unstable
+    public NotificationEmailDiffType getDiffType(UserReference user)
+    {
+        return emailUserPreferenceManager.getDiffType(user);
+    }
+
+    /**
+     * @return the email notification interval configured for the current user
+     * @since 14.10
      */
-    public NotificationEmailDiffType getDiffType(String userId)
```

Comment:
So same here: you should have kept that one but deprecate it.

Reply:
Same here you're missing the `@Deprecated` annotation on the method (besides the javadoc one)

Reply:
Sounds like you misunderstood me, there's two different things to have: a `@deprecated` javadoc documentation tag, and a Java `@Deprecated` annotation that is used for our tooling. See for example: https://github.com/xwiki/xwiki-platform/blob/master/xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/store/XWikiStoreInterface.java#L111

Reply:
Right, sorry for the misunderstanding, I knew there was something wrong, I searched for "@deprecated" lower case only. Fixed it. Many thanks for your guidance.